src/components/date/date.pw.tsx

```diff
@@ -0,0 +1,1349 @@
+    const wrapper = page.locator('div[class="DayPicker-wrapper"]');
+    await expect(wrapper).toHaveCount(1);
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Escape");
+    await expect(wrapper).toHaveCount(0);
+    await expect(getDataElementByValue(page, "input")).toBeFocused();
+  });
+
+  test(`should close the picker when shift + tab is pressed and focus is on the previous month button in the picker and refocus the input`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DateInputCustom />);
+
+    await page.focus("body");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    const wrapper = page.locator('div[class="DayPicker-wrapper"]');
+    await expect(wrapper).toHaveCount(1);
+    await page.keyboard.press("Shift+Tab");
+    await expect(wrapper).toHaveCount(0);
+    await expect(getDataElementByValue(page, "input")).toBeFocused();
+  });
+
+  test(`should navigate to the next month when right arrow pressed on last day element of a month`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<DateInputCustom value="31/05/2022" />);
+
+    await page.focus("body");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press("Tab");
+    await page.keyboard.press(arrowKeys[0]);
+    const focusedElement = page
+      .getByRole("row")
+      .nth(1)
+      .locator("div")
+      .filter({ hasText: "1" })
+      .filter({ hasNotText: "31" });
+    await expect(focusedElement).toBeFocused();
+    const pickerHeading = page.locator(".DayPicker-Caption").locator("div");
+    await expect(pickerHeading).toHaveText(NEXT_MONTH);
+  });
+
+  [
+    ["Enter", "next", "chevron_right"],
+    ["Space", "next", "chevron_right"],
+    ["Enter", "previous", "chevron_left"],
+    ["Space", "previous", "chevron_left"],
+  ].forEach(([key, month, arrow]) => {
+    test(`should change the displayed month when ${key} is pressed and ${month} button is focused`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<DateInputCustom />);
+
+      const inputParent = getDataElementByValue(page, "input").locator("..");
+      await inputParent.click();
+      const arrowParent = getDataElementByValue(page, arrow).locator("..");
+      await arrowParent.focus();
+      const arrowElement = getDataElementByValue(page, arrow);
+      await arrowElement.press(key);
+
+      const pickerHeading = page.locator(".DayPicker-Caption").locator("div");
+      if (month === "next") {
+        await expect(pickerHeading).toHaveText(NEXT_MONTH);
+      } else if (month === "previous") {
+        await expect(pickerHeading).toHaveText(PREVIOUS_MONTH);
+      }
+    });
+  });
+
+  ([
+    [SIZE.SMALL, 30],
+    [SIZE.MEDIUM, 38],
+    [SIZE.LARGE, 46],
+  ] as [DateInputProps["size"], number][]).forEach(([size, approxValue]) => {
+    test(`should check the ${size} of the DateInput`, async ({
```

Comment:
```suggestion
    test(`should check the ${size} size of the DateInput`, async ({
```

**suggestion** - again I know this wasn't you, but as it is this will just output eg "should check the small of the DateInput"` as a test description